### ISSUE TITLE
Switch from TSLint to ESLint

### ISSUE THREAD

**Issue by MichalZalecki** [codechecks/monorepo#38]
@krzkaczor I would like to know your thoughts on switching from TSLint to ESLint due to the [planned depreciation](https://github.com/palantir/tslint/issues/4534).

**Comment by krzkaczor** [comment]
I am totally happy to move to ESLint. The only reason why I haven't do it already is that I think there is a lack of ESLint counterparts for more complicated TSlint rules. Things like `no-floating-promises` etc. But I think I am out of loop here and probably there was a lot of progress in the meantime...

We use [TypeStrict](https://www.npmjs.com/package/typestrict) which could be ported to ESLint.

**Comment by carmanchris31** [comment]
Indeed, a lot of progress has been made with a lot of collaboration from the TSLint creator.
`no-floating-promises` now exists as a rule here: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md
Here is a list of mappings from tslint to their best-equivalents: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md
